### ISSUE TITLE
Allocated a negative number of bytes

### ISSUE THREAD

**Issue by J2-Tech** [SquidDev-CC/CC-Tweaked#1739]
### Minecraft Version

1.20.x

### Version

1.109.6

### Details

Getting some spam on my arclight server logs : 

Ex: 
[05:25:28 WARN] [d.c.c.c.c.ComputerThread]: Allocated a negative number of bytes (-3093480)!
[06:52:38 WARN] [d.c.c.c.c.ComputerThread]: Allocated a negative number of bytes (-3241872)!
[07:08:06 WARN] [d.c.c.c.c.ComputerThread]: Allocated a negative number of bytes (-3219152)!
[07:54:31 WARN] [d.c.c.c.c.ComputerThread]: Allocated a negative number of bytes (-2905632)!
[08:29:22 WARN] [d.c.c.c.c.ComputerThread]: Allocated a negative number of bytes (-3221248)!

here are part of my server logs 
[log-part.log](https://github.com/cc-tweaked/CC-Tweaked/files/14539071/log-part.log)

**Comment by J2-Tech** [comment]
Created after commenting on #1672

**Comment by SquidDev** [comment]
Would you be able to upload your full logs? They tend to include some other information that is useful for debugging!

**Comment by J2-Tech** [comment]
Yes, here they are. i wanted to make sure no ips or usernames were in there, but i think it's OK
[debug.log](https://github.com/cc-tweaked/CC-Tweaked/files/14540252/debug.log)

**Comment by SquidDev** [comment]
Yeah, this is all very odd - I cannot see any reason why this would happen. I've added some additional logging in the latest CC:T release. Would you be able to update to 1.110.0 and upload your logs again? Thanks, and sorry!

**Comment by TheStraying11** [comment]
https://mclo.gs/u7wI7TN#L6007 this was taken with 1.110.1

**Comment by J2-Tech** [comment]
[debug.log](https://github.com/cc-tweaked/CC-Tweaked/files/15177233/debug.log)
A little late on this but here is the debug log with the extra logging you've added

**Comment by SquidDev** [comment]
Thanks for the extra info both!

```
Allocated a negative number of bytes (-2667800)!
    Previous measurement at t=2667975172513365 on Thread #203 = 22172296600
     Current measurement at t=2667975172784822 on Thread #203 = 22169628800
Allocated a negative number of bytes (-3753488)!
    Previous measurement at t=2672289821893707 on Thread #203 = 59493198512
     Current measurement at t=2672289822061122 on Thread #203 = 59489445024
Allocated a negative number of bytes (-4186952)!
    Previous measurement at t=2674160200447684 on Thread #203 = 81195106984
     Current measurement at t=2674160200587238 on Thread #203 = 81190920032
Allocated a negative number of bytes (-454672)!
    Previous measurement at t=3513475356971 on Thread #174 = 982904520
     Current measurement at t=3513475583092 on Thread #174 = 982449848
```

This is very odd. We're definitely not measuring the wrong thread, so the allocation counter is sometimes going backwards. The two measurements are between 100µs and 300µs apart, which is short, but not so short I'd expect it to be an issue.

Taking a look at the implementation of [`Thread::cooked_allocated_bytes`](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/thread.inline.hpp#L38-L59), there are several cases where the value might decrease (for instance, when fetching this value while the TLAB is being cleared). But this is happening surprisingly often for what feels like it should be a rare race condition.

**Comment by J2-Tech** [comment]
Here are two of the main programs running on multiple turtles / computers 
https://pastebin.com/G6CGbhAt

https://pastebin.com/1Lgvt4dz


I'm also running This on two or three computers with massive ammounts of storage connected to the network.

https://github.com/SquidDev-CC/artist

**Comment by rugter100** [comment]
I seem to also be having this error in my logs. Not sure what exactly could be causing it since the programs the computers are running are made by a friend of mine. As far as i know we only have pc's interacting with create train stations and such.

**Comment by MistakingManx** [comment]
I'm also having this issue, with increased storage limits. But the only programs running are mine doing basic things in while loops that run every second.

**Comment by TheCakeOfRice** [comment]
+1 to this.  I have a [storage system](https://github.com/TheCakeOfRice/computercraft) that uses rednet and http calls heavily.  The warnings start showing in the logs as soon as cpus are turned on, and the negative values get bigger over time.  Rednet and http APIs stop functioning eventually.
